core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcRules.java

```diff
@@ -466,12 +470,11 @@ public JdbcProjectRule(final JdbcConvention out) {
     }
 
     /** Creates a JdbcProjectRule. */
-    public JdbcProjectRule(final JdbcConvention out,
+    public  JdbcProjectRule(final JdbcConvention out,
         RelBuilderFactory relBuilderFactory) {
       super(Project.class, (Predicate<Project>) project ->
-              (out.dialect.supportsWindowFunctions()
-                  || !RexOver.containsOver(project.getProjects(), null))
-                  && !userDefinedFunctionInProject(project),
+              out.dialect.supportsWindowFunctions()
+                  || !RexOver.containsOver(project.getProjects(), null),
```

Comment:
This is probably ok for now, but a bit too loose in general. It'll try to push too much